### ISSUE TITLE
quotemark rule should not apply to JSON files

### ISSUE THREAD

**Issue by JamesMessinger** [palantir/tslint#4089]
### Bug Report

- __TSLint version__:  5.11.0
- __TypeScript version__:  3.0.1
- __Running TSLint via__: CLI

#### TypeScript code being linted

**sample.ts**
```ts
// TypeScript now allows you to import JSON files
import data from 'data.json';
console.log(data)
```

**data.json**
```json
{
    "foo": "bar"
}
```

**tslint.yaml**
```yaml
rules:
  quotemark:
    - true
    - single
```

#### Actual behavior

When running `tslint --fix`, the double quotes in the `data.json` file are replaced with single-quotes due to the `quotemark` rule.  But single-quotes are not valid in JSON files.  Only double quotes are valid JSON.

#### Expected behavior

The `quotemark` rule should not apply to JSON files.

**Comment by adidahiya** [comment]
Let's skip linting JSON files altogether rather than address this issue for only one rule.  #4426

**Comment by adidahiya** [comment]
Just to clarify, you are importing JSON files using [the TS 2.9+ `--resolveJsonModule` option](https://github.com/Microsoft/TypeScript/wiki/What%27s-new-in-TypeScript#new---resolvejsonmodule), right?

**Comment by JamesMessinger** [comment]
Yes, that's correct.

**Comment by adidahiya** [comment]
should be fixed in v5.12.1, which no longer tries to lint JSON files at all. let me know if that works for you

**Comment by JamesMessinger** [comment]
Awesome!  thanks!
